src/hooks/shapediver/viewer/useOutputDrawingTools.ts

```diff
@@ -0,0 +1,99 @@
+import { IGeometryData, IOutputApi, ISessionApi, ITreeNode, IViewportApi } from "@shapediver/viewer";
+import { useCallback, useEffect, useRef } from "react";
+import { IDrawingToolsApi, PlaneRestrictionProperties, RESTRICTION_TYPE, createDrawingTools } from "@shapediver/viewer.features.drawing-tools";
+import { useShapeDiverStoreViewer } from "store/useShapeDiverStoreViewer";
+import { CustomizationPropertiesOptional } from "@shapediver/viewer.features.drawing-tools/dist/business/implementation/DrawingToolsManager";
+import { useOutputNode } from "./useOutputNode";
+
+/**
+ * Hook allowing to create the drawing tools.
+ * 
+ * Makes use of {@link useOutputNode}.
+ * 
+ * @param sessionId 
+ * @param viewportId 
+ * @param outputIdOrName 
+ */
+export function useOutputDrawingTools(sessionId: string, viewportId: string, outputIdOrName: string) : {
+	/**
+	 * API of the output
+	 * @see https://viewer.shapediver.com/v3/latest/api/interfaces/IOutputApi.html
+	 */
+	outputApi: IOutputApi | undefined,
+
+	drawingToolsApiRef: React.MutableRefObject<IDrawingToolsApi | null>
+} {
+
+	/**
+	 * Create the callbacks for the drawing tools
+	 * @ALEX I'm not sure how to actually get parameters, if not via the API
+	 */
+	const onFinish = async (geometryData: IGeometryData) => {
```

Comment:
@MajorMeerkatThe3rd for defining callbacks, always use the [useCallback](https://react.dev/reference/react/useCallback) hook.